### ISSUE TITLE
fatal error: bracket nesting level exceeded maximum of 256

### ISSUE THREAD

**Issue by johnjelinek** [crosstool-ng/crosstool-ng#397]
Is there a config to increase the bracket nesting level `CFLAGS`?
```
[INFO ]  Installing pass-1 core C gcc compiler
[EXTRA]    Configuring core C gcc compiler
[EXTRA]    Building gcc
[ERROR]    /Volumes/CrossTool2NG/arm-unknown-linux-gnueabi/.build/src/gcc-5.2.0/gcc/config/arm/neon.md:3463:10917: fatal error: bracket nesting level exceeded maximum of 256
[ERROR]    make[3]: *** [insn-attrtab.o] Error 1
[ERROR]    make[2]: *** [all-gcc] Error 2
[ERROR]
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Installing pass-1 core C gcc compiler'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@257]
[ERROR]  >>        called from: do_gcc_core_backend[scripts/build/cc/100-gcc.sh@537]
[ERROR]  >>        called from: do_gcc_core_pass_1[scripts/build/cc/100-gcc.sh@112]
[ERROR]  >>        called from: do_cc_core_pass_1[scripts/build/cc.sh@35]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@646]
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      '/Volumes/CrossTool2NG/crosstool-ng-1.22.0/docs/B - Known issues.txt'
[ERROR]
[ERROR]  (elapsed: 43:48.00)
[43:48] / make[1]: *** [build] Error 2
make: *** [build.4] Error 2
```

**Comment by johnjelinek** [comment]
```
CFLAGS="-O2 -fbracket-depth=1024" ../crosstool-ng-1.22.0/ct-ng build.4
[INFO ]  Performing some trivial sanity checks
[ERROR]  Don't set CFLAGS. It screws up the build.
[ERROR]
[ERROR]  >>
[ERROR]  >>  Build failed in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_Abort[scripts/functions@331]
[ERROR]  >>        called from: CT_TestAndAbort[scripts/functions@351]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@58]
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      '/Volumes/CrossTool2NG/crosstool-ng-1.22.0/docs/B - Known issues.txt'
[ERROR]
[ERROR]  (elapsed: 24406411:06.00)
[00:00] / make[1]: *** [build] Error 1
make: *** [build.4] Error 2
```

**Comment by johnjelinek** [comment]
I modified the CFLAGS in source `scripts/build/cc/100-gcc.sh`

**Comment by bhundven** [comment]
Yea, this is another thing that is fixed in what will be 1.23.

**Comment by MarkMMullin** [comment]
I didn't see this in the issues list and to me I interpret it as 'if you want to run the compiler on the Mac, you can't'  Did I miss the issue ?   I'd really like to be able to compile on the Mac and deliver to the Beaglebone, I have to rebuild the whole thing and stuff Xenomai into it  :-(

**Comment by DanDevine** [comment]
Same issue here, looking for workaround:
```
[ALL  ]    /Volumes/rpi-big/.build/src/gcc-5.2.0/gcc/config/arm/arm.md:78:16: note: use '=' to turn this equality comparison into an assignment
[ALL  ]    (thumb1_code)) == (
[ALL  ]                   ^~
[ALL  ]                   =
[ERROR]    /Volumes/rpi-big/.build/src/gcc-5.2.0/gcc/config/arm/neon.md:3463:10917: fatal error: bracket nesting level exceeded maximum of 256
[ALL  ]    /Volumes/rpi-big/.build/src/gcc-5.2.0/gcc/config/arm/neon.md:3463:10917: note: use -fbracket-depth=N to increase maximum nesting level
[ALL  ]    47 warnings and 1 error generated.
[ERROR]    make[2]: *** [insn-attrtab.o] Error 1
[ALL  ]    make[2]: *** Waiting for unfinished jobs....
[ALL  ]    44 warnings generated.
[ALL  ]    18 warnings generated.
[ALL  ]    18 warnings generated.
[ALL  ]    42 warnings generated.
[ALL  ]    rm gcc.pod
[ERROR]    make[1]: *** [all-gcc] Error 2
[ERROR]  
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Installing pass-1 core C gcc compiler'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@216]
[ERROR]  >>        called from: do_gcc_core_backend[scripts/build/cc/100-gcc.sh@537]
[ERROR]  >>        called from: do_gcc_core_pass_1[scripts/build/cc/100-gcc.sh@112]
[ERROR]  >>        called from: do_cc_core_pass_1[scripts/build/cc.sh@35]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@646]
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      '/usr/local/Cellar/crosstool-ng/1.22.0_1/share/doc/crosstool-ng/crosstool-ng-1.22.0/B - Known issues.txt'
[ERROR]  
[ERROR]  (elapsed: 35:35.00)
```

When attempting to add the '-fbracket-depth=512' modifier in menuconfig, I get error similar to above...


```
[INFO ]  Installing ncurses for build
[DEBUG]    Entering '/Volumes/rpi-big/.build/armv7-rpi2-linux-gnueabihf/build/build-ncurses-build-x86_64-build_apple-darwin15.6.0'
[EXTRA]    Configuring ncurses
[DEBUG]    ==> Executing: 'CFLAGS=-O2 -g -fbracket-depth=512' 'LDFLAGS= ' '/Volumes/rpi-big/.build/src/ncurses-6.0/configure' '--build=x86_64-build_apple-darwin15.6.0' '--host=x86_64-build_apple-darwin15.6.0' '--prefix=' '--with-install-prefix=/Volumes/rpi-big/.build/armv7-rpi2-linux-gnueabihf/buildtools' '--enable-termcap' '--enable-symlinks' '--without-manpages' '--without-tests' '--without-cxx' '--without-cxx-binding' '--without-ada' '--with-abi-version=5' 
[CFG  ]    checking for egrep... grep -E
[CFG  ]    Configuring NCURSES 6.0 ABI 6 (Sun Nov 27 02:21:48 PST 2016)
[CFG  ]    checking for package version... 6.0
[CFG  ]    checking for package patch date... 20151107
[CFG  ]    configure: WARNING: overriding ABI version 6 to 5
[CFG  ]    checking build system type... x86_64-build_apple-darwin15.6.0
[CFG  ]    checking host system type... x86_64-build_apple-darwin15.6.0
[CFG  ]    checking target system type... x86_64-build_apple-darwin15.6.0
[CFG  ]    Configuring for darwin15.6.0
[CFG  ]    checking for prefix... 
[CFG  ]    checking for x86_64-build_apple-darwin15.6.0-gcc... x86_64-build_apple-darwin15.6.0-gcc
[ERROR]    checking for C compiler default output... configure: error: C compiler cannot create executables
[ERROR]  
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Installing ncurses for build'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@216]
[ERROR]  >>        called from: do_ncurses_backend[scripts/build/companion_libs/220-ncurses.sh@130]
[ERROR]  >>        called from: do_ncurses_for_build[scripts/build/companion_libs/220-ncurses.sh@35]
[ERROR]  >>        called from: do_companion_libs_for_build[scripts/build/companion_libs.sh@29]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@646]
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      '/usr/local/Cellar/crosstool-ng/1.22.0_1/share/doc/crosstool-ng/crosstool-ng-1.22.0/B - Known issues.txt'
[ERROR]  
[ERROR]  (elapsed: 0:03.00)
```



Where did you modify the '100-gcc.sh' script, and did it actually work?

Wishlist item:  Path variable in 'menuconfig-->paths' where user could specify location GCC to use during initial construction (for example, if installed via homebrew or other alternate means)

Thanks

**Comment by stilor** [comment]
@DanDevine Why does it have to be a menuconfig item? Isn't setting `$PATH` to include the directory for that custom GCC location working?

**Comment by DanDevine** [comment]
@stilor Not required, just helpful in avoiding the constant switching back and forth...when installing GCC on OS X via home-brew, I needed to modify the $PATH to have '/usr/local/bin' first...AND needed to create a symlink `ln -s ../Cellar/gcc/6.2.0/bin/gcc-6 gcc`.   Then, after creating this, need to reload the configuration...because the OS X terminal seems to pick-up which gcc I'm using on initialization.  I'm really just looking for an easier way to explicitly select which host tool set I'm using when building the cross compiler.  Would be easier to install and compartmentalize the various overlapping versions of various OS X related stuff.

ps:  still having the 'bracket-depth' issue regardless of which host compiler I'm using (Apple LLVM or GCC 6.2.0).  I patched in the few lines around this issue from https://github.com/crosstool-ng/crosstool-ng/commit/b0743fdcda5e203c5b379d6b78a8adb74c4084ff#diff-527dce9a71f9422aa473e78aff03ae2c and it doesn't seem to make any difference.

**Comment by walsh4853** [comment]
I have exactly the same issue.

Is there a resolution or a work-around?

**Comment by shwnchpl** [comment]
Same issue here as well.  Doesn't seem to be any way that I can actually pass the -fbracket-depth=1024 flag to where it needs to be passed to; no matter where I put it, I end up getting the error about exceeding a depth of 256.

**Comment by stilor** [comment]
To all who commented after the bug was closed: please indicate if you tried 1.22 or master. Current master passes -fbracket-depth=512 while building with clang.

**Comment by Kochise** [comment]
LLVM 6.0.0 x64 win32 :

1>main_tst.c(256):  note: use -fbracket-depth=N to increase maximum nesting level
1>clang-cl.exe : warning : unknown argument ignored in clang-cl: '-fbracket-depth=512' [-Wunknown-argument]

Dudes, in the same output log, seriously...
